readme.md

```diff
@@ -23,3 +23,262 @@ For the revision task, please refer to: [revision.md](revision.md).
+    |  `>=`      |Greater than or equal to |  7 >= 3 ~ 1   |
+    |   `\<=`    |Smaller than or equal to| 7 \<= 3 ~ 0    |
+    |   `!=`     |Different  |  7 != 3 ~ 1   |
+- **Logical operator\:**
+  - **Logical AND (`&&`)\:** is a Boolean operation that evaluates two conditions to determine whether both are true. The result is true only if both operands are true; otherwise, it is false.
+   ![alt text](image-5-1.png)
+  - **Logical OR(`||`)\:** return the right-hand side operand if the left one is non-zero. Otherwise, it returns the left-hand side itself.
+  
+    ![alt text](image-6-1.png)
+
+  - **Operator NOT (`!`)\:** return 1 when  the operand is FALSE and vice versa.
+- **Bitwise operator\:**
+  - **Bitwise AND(`&`)\:** used to perform a bitwise AND operation between the corresponding bits of two integer operands by comparing the bits at each position. It returns an integer result where each bit position contains the result of the OR operation on the corresponding bits of the operands.
+ ![alt text](image-10.png)
+   - **Bitwise OR(`|`)\:** used to perform a bitwise OR operation between the corresponding bits of two integer operands by comparing the bits at each position. It returns an integer result where each bit position contains the result of the OR operation on the corresponding bits of the operands.
+  ![alt text](image-9.png)
+  - **Bitwise XOR(`^`)\:** return 1 if the corresponding bits of two operands are opposite.
+  ![alt text](image-11.png)
+  - **Bitwise complement\:** a unary operator works on only one operand by changing 1 to 0 and 0 to 1.
+  ![alt text](image-12.png)
+  - **Shift operator\:**
+    - Right shift operator (`>>`): shifts all bits towards right by certain number of specified bits.
+    ![alt text](image-13.png)
+    - Left shift operator(`<<`): shifts all bits towards left by certain number of specified bits.
+    ![alt text](image-14.png)
+- **Difference between logical AND and bitwise AND\:**
+
+| Aspect           | Bitwise AND (`&`)                                            | Logical AND (`&&`)                                          |
+| ---------------- | ------------------------------------------------------------ | ----------------------------------------------------------- |
+| Operand Type     | Operates on integer operands (integral types).               | Operates on boolean operands (`int`, `char`, `bool`, etc.). |
+| Usage            | Performs bitwise AND operation between corresponding bits.   | Performs logical AND operation between boolean expressions. |
+| Result           | Returns an integer with bits set based on the AND operation. | Returns a boolean (`true` or `false`) based on logical AND. |
+| Short-circuiting | Does not perform short-circuit evaluation.                   | Performs short-circuit evaluation.                          |
+| Evaluation       | Both operands are always evaluated.                          | Right operand is only evaluated if left operand is `true`.  |
```

Comment:
@spm1102 
`#reply_required`
`#for_information`

To optimize the program:

- The conditions which rarely happens should be placed in the left side of `&&` or `||`: If those conditions are false regularly, the other conditions in the right side will not need to be computed.
- The complex conditions should be placed in the right side of `&&` or `||` : If the condition in the left side is true, the complex conditions will not need to be computed.

Reply:
@Dark-GreenCat Oh great! this is the application of short-circuit, isn't it?

Reply:
You got it!